Source/JavaScriptCore/jsc.cpp

```diff
@@ -3937,15 +3938,16 @@ static NEVER_INLINE void crashPGMUpperGuardPage()
 {
     WTF::forceEnablePGM();
     size_t allocSize = getpagesize() * 10000;
-    char* result = static_cast<char*>(fastMalloc(10000000));
+    char* result = static_cast<char*>(fastMalloc(allocSize));
     result = result + allocSize;
     *result = 'a';
 }
 
 static NEVER_INLINE void crashPGMLowerGuardPage()
 {
     WTF::forceEnablePGM();
-    char* result = static_cast<char*>(fastMalloc(10000000));
+    size_t allocSize = getpagesize() * 10000;
+    char* result = static_cast<char*>(fastMalloc(allocSize));
     result = result - getpagesize();
     *result = 'a';
```

Comment:
I think we should change this to write one byte before the start of the allocation instead of a whole page.  (This matches `crashPGMUpperGuardPage()` which writes one byte past the end of the allocation.)

I think subtracting 1 will work here:

```suggestion
    char* result = static_cast<char*>(fastMalloc(allocSize));
    result = result - 1;
    *result = 'a';
```

Reply:
That sounds good. Will try that tomorrow.